File: ProyectoPoo/ProyectoPooJava/src/ec/edu/utpl/semana9/Fraccion.java
package ec.edu.utpl.semana9;

public class Fraccion {
    public static int catidadDeNumeradores=1;//Iniciar Instancia
    public static int cantidadDeDenominadores=1;
    private int numerador;//Atributos
    private int denominador;//Atributos

    public Fraccion() {//Clase
        this.setNumerador(0);
        this.setDenominador(1);
    }

    public Fraccion(int numerador, int denominador) {//Constructor
        this.setNumerador(numerador);
        if( denominador == 0) {
            throw new IllegalArgumentException("Denominador no pueder ser 0 o negativo");//Validar datos
        }
        this.setDenominador(denominador);
    }
    public Fraccion(int numerador) {//Sobrecarga
        this.numerador = numerador;
        this.denominador = 1;
    }

    public int getNumerador() {//Inicio de los get y set
        return numerador;
    }

    public void setNumerador(int numerador) {
        this.numerador = numerador;
    }

    public int getDenominador() {
        return denominador;
    }

    public void setDenominador(int denominador) {
        this.denominador = denominador;
    }//Fin de los get y set
    public String toString() {
        return numerador + "/" + denominador;
    }
    public Fraccion dividir(Fraccion f) {
        Fraccion aux = new Fraccion();
        aux.numerador = this.numerador * f.denominador;
        aux.denominador = this.denominador * f.numerador;
        return aux;
    }
}

File: ProyectoPoo/ProyectPooKotlin/src/ec/edu/utpl/semana9/Fraccion.kt
package ec.edu.utpl.semana9

class Fraccion {
    private var numerador //Atributos
            = 0
    private var denominador //Atributos
            = 0

    constructor() { //Clase
        setNumerador(0)
        setDenominador(1)
    }

    constructor(numerador: Int, denominador: Int) { //Constructor
        setNumerador(numerador)
        require(denominador != 0) {
            "Denominador no pueder ser 0 o negativo" //Validar datos
        }
        setDenominador(denominador)
    }

    constructor(numerador: Int) { //Sobrecarga
        this.numerador = numerador
        denominador = 1
    }

    fun getNumerador(): Int { //Inicio de los get y set
        return numerador
    }

    fun setNumerador(numerador: Int) {
        this.numerador = numerador
    }

    fun getDenominador(): Int {
        return denominador
    }

    fun setDenominador(denominador: Int) {
        this.denominador = denominador
    } //Fin de los get y set

    override fun toString(): String {
        return "$numerador/$denominador"
    }

    fun dividir(f: Fraccion): Fraccion {
        val aux = Fraccion()
        aux.numerador = numerador * f.denominador
        aux.denominador = denominador * f.numerador
        return aux
    }

    companion object {
        var catidadDeNumeradores = 1 //Iniciar Instancia
        var cantidadDeDenominadores = 1
    }
}